### PR TITLE
Initial data layer and data query component for `/all-domains` endpoint

### PR DESCRIPTION
**Pull request by olaseni**
#### Changes proposed in this Pull Request

*  Gets domains data from a new HTTP path  i.e. `/all-domains`. 
*  A new data component `QueryAllDomains` to manage fetching domains data. 

#### Testing instructions

* Start your Calypso dev server and navigate to : http://calypso.localhost:3000/domains/manage
* Use the Redux profiler to ensure that a new key `allDomains` exists in the tree.

### CODE REVIEW

File: client/state/data-layer/wpcom/all-domains/index.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+		errorNotice( translate( 'Failed to retrieve all domains' ) ),
+		getAllDomainsRequestFailure( error ),
+	];
+};
+
+export const getAllDomainsSuccess = ( action, response ) => {
+	if ( response ) {
+		return getAllDomainsRequestSuccess( response.domains );
+	}
+	return getAllDomainsError( action, 'Failed to retrieve your domains. No response was received' );
```
Comment (hambai) [inline client/state/data-layer/wpcom/all-domains/index.js:41]: I know it shouldn't be possible to end up in that situation but maybe we should not handle it as an error - not getting any domains might also mean that you have no domains. Technically you should not be able to access the "all my sites" page in that case but who knows what will happen... :D I'm fine though to fix this in a follow up PR

Comment (olaseni) [reply client/state/data-layer/wpcom/all-domains/index.js:41]: In such a case the **_response_** from the endpoint should _return_ an explicit empty array.  That is clearer to wrangle. Not receiving a response could be as a result of a timeout.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/3763827

Hi @olaseni, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include this string: `Failed to retrieve all domains`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/3763827).

**Pull request merged**